service/src/main/java/uk/gov/hmcts/reform/fpl/service/children/ChildrenDataFixer.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.hmcts.reform.fpl.service.children;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.Child;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+
+import java.util.List;
+import java.util.Map;
+
+import static uk.gov.hmcts.reform.fpl.utils.CaseDetailsHelper.isInOpenState;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDetailsHelper.isInReturnedState;
+
+@Component
+public class ChildrenDataFixer {
+
+    private final ObjectMapper mapper;
+
+    @Autowired
+    public ChildrenDataFixer(ObjectMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    public CaseDetails fix(final CaseDetails caseDetails) {
```

Comment:
minor, from method signature it is hard to understand what it does, need to read method body to get this knowledge